### ISSUE TITLE
XPath-issue with ows:Exception

### ISSUE THREAD

**Issue by matsoo** [opengeospatial/ets-wfs11#66]
When I test the new basic tests on the beta site I encounter an issue in the wfs:wfs-1.1.0-Basic-DescribeFeatureType-tc5 and wfs:wfs-1.1.0-Basic-GetFeature-POST-tc6. I think the problem is that both test try to send a request without a version parameter, and our server then returns a Exception with the newer ows 1.1 namespace since we also support WFS 2.0. So the exception use the namespace http://www.opengis.net/ows/1.1 while the test expects http://www.opengis.net/ows.

**Comment by dstenger** [comment]
Thank you for reporting.

The reported issue is similar to following bug in the ets-wcs11: https://github.com/opengeospatial/ets-wcs11/issues/20

Can you please check what the WFS 1.1 specification says about this case? What response is expected when the version parameter is missing?

**Comment by matsoo** [comment]
It is not obvious from the specification. I know this issue has been discussed before which is the reason we went with returning the latest version in the first place. 
The problem is that both WFS 1.1.0 and 2.0 demand you use the correct version of the exception report when encountering an invalid parameter (neither mentions what to do with a missing version parameter specifically), but both also supports having multiple versions on the same endpoint. 
So the problem is what version are you really requesting when not sending the version parameter? I think this really needs to be clarified in future OWS versions. 
Sending the latest version seems logical to me and aligns with how version negotiation for capabilities is handled, but as far as I am aware it is not really specified anywhere how to handle requests with missing version parameters other than capabilities.

**Comment by dstenger** [comment]
As both tests do not test the missing version parameter itself but just if an invalid request returns the correct exception, I propose that both tests should be adjusted by not removing the version parameter but any other mandatory parameter.

Then, the reported problem does not occur any more.

**Comment by bermud** [comment]
@matsoo, are you ok with dstenger solution?

**Comment by lgoltz** [comment]
Basically I agree with the solution of @dstenger. But in the case of the test  **wfs:wfs-1.1.0-Basic-DescribeFeatureType-tc5** it is a bit difficult, cause there are no other mandatory parameter. The test consist of two parts as explained in the description:
 ```
DescribeFeatureType with 
  [1] missing service parameter, 
  [2] missing version parameter. 
Pass if all of  the following conditions are true: (1) the response is schema valid; (2) the root document is an ows:ExceptionReport document.
```
I would suggest to remove part [2] completely from the test.

In the test **wfs:wfs-1.1.0-Basic-GetFeature-POST-tc6** I can't see that the version parameters is missing. I've tried with the ETS "Web Feature Service (WFS) 1.1.0, Version 1.25" on [1] against the deegree reference implementation [2]. Here are the details from the test result: [ets-wfs-11_wfs-1.1.0-Basic-GetFeature-POST-tc6.txt](https://github.com/opengeospatial/ets-wfs11/files/538969/ets-wfs-11_wfs-1.1.0-Basic-GetFeature-POST-tc6.txt)
In both test steps the version parameter is set. The test results in an exception (as expected), cause
1. an undefined feature type is requested.
2. an undefined property is requested.

[1] http://cite.opengeospatial.org/teamengine/
[2] http://cite.deegree.org/deegree-webservices-3.4-RC1/services/wfs110?service=WFS&request=GetCapabilities

**Comment by dstenger** [comment]
I agree with the analysis of @lgoltz.

Regarding test **wfs:wfs-1.1.0-Basic-DescribeFeatureType-tc5** the second part of the test can be removed as it basically tests the same functionality like the first part (just a different mandatory parameter is removed).

@matsoo can you please specify your problems with test **wfs:wfs-1.1.0-Basic-GetFeature-POST-tc6**?

**Comment by matsoo** [comment]
I thought it was the same problem as the first test but when I look at it again I realize it is not, so you can ignore that part. Otherwise I agree with your solution.

**Comment by dstenger** [comment]
Fix is provided by pull request #67.
It is reviewed and will be included in the next release of the ETS.

**Comment by dstenger** [comment]
The fix was tested successfully, merged into master branch and will be part of the next release.
